includes/derivatives.inc

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * @file
+ * Holds derivative functionality for compound solution pack.
+ */
+
+/**
+ * Derivative creation function for compound parent object TN.
+ *
+ * Wraps the TN update/creation function and
+ * adds expected logging.
+ *
+ * @param AbstractObject $object
+ *   The compound object.
+ * @param bool $force
+ *   Boolean for forcing this derivative creation.
+ *
+ * @return array
+ *   Success message.
+ */
+function islandora_compound_object_create_thumbnail(AbstractObject $object, $force = FALSE) {
+  $parts = islandora_compound_object_get_parts($object->id);
+  if (!empty($parts)) {
+    $child = islandora_object_load($parts[0]);
+    islandora_compound_object_update_parent_thumbnail($object, $child);
+  }
+  if (isset($child) && $object['TN'] == $child['TN']) {
+    return array(
+      'success' => TRUE,
+      'messages' => array(
+      array(
+        'message' => t('Updated @dsid on @parent.'),
+        'message_sub' => array(
+          '@dsid' => 'TN',
+          '@parent' => $object->id,
+        ),
+        'type' => 'dsm',
+        'severity' => 'status',
+      ),
+      ),
+      'file' => 'islandora_compound_object.module',
```

Comment:
I... think you might've misread [comment in the documentation](https://github.com/Islandora/islandora/blob/7.x/islandora.api.php#L691-L702) (applies both here and below).